azuredevops/internal/service/servicehook/commons.go

```diff
@@ -0,0 +1,135 @@
+package servicehook
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/servicehooks"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+)
+
+const (
+	errMsgTfConfigRead       = "Reading terraform configuration: %+v"
+	errMsgSubscriptionRead   = "Looking up service hook subscription given ID (%s): %v"
+	errMsgSubscriptionDelete = "Deleting service hook subscription. SubscriptionID: %s. %v"
+)
+
+// baseSchema returns the base schema elements that are common to all service hook resources
+func baseSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"project_id": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IsUUID,
+			Description:  "The ID of the project where the service hook subscription will be created",
+		},
+	}
+}
+
+// createSubscription creates a new service hook subscription in Azure DevOps
+func createSubscription(d *schema.ResourceData, clients *client.AggregatedClient, subscription *servicehooks.Subscription) (*servicehooks.Subscription, error) {
```

Comment:
All these CRUD wrapper functions do nothing more than simply calling the underlying client's corresponding method, I prefer to just inlining them at where they are invoked, instead of adding this unnecessary layer.